### PR TITLE
perf: mentioned type and size of the doc to be uploaded

### PR DESCRIPTION
**Pull request by sumitbishti**
explicitly mentioned "PDF" to upload, and added a toast if pdf size is greater than 50mb

fixes: #621

### CODE REVIEW

File: apps/web/src/app/(dashboard)/documents/upload-document.tsx [hunk 1]
```diff
@@ -96,13 +103,23 @@ export const UploadDocument = ({ className }: UploadDocumentProps) => {
     }
   };
 
+  const onFileDropRejected = () => {
+    toast({
+      title: 'File is larger than 50mb',
```
Comment (dguyen) [inline apps/web/src/app/(dashboard)/documents/upload-document.tsx:108]: Since this limit actually depends on the provider you use, could you create the following environment variable so users can dynamically change it for their use case?

```
# OPTIONAL: Displays the maximum document upload limit to the user in MBs
NEXT_PUBLIC_DOCUMENT_UPLOAD_LIMIT=5
```

Also could you please reword it to be `File cannot be larger than ...`

**Review by dguyen — CHANGES_REQUESTED** [review]

File: apps/web/src/app/(dashboard)/documents/upload-document.tsx (hunk 1 above)
Comment (sumitbishti) [reply apps/web/src/app/(dashboard)/documents/upload-document.tsx:108]: Should i also use this env variable in the codebase in place of the limit 50mb? Or just create the env var and leave it like that?

Comment (dguyen) [reply apps/web/src/app/(dashboard)/documents/upload-document.tsx:108]: Yeah that'll be a good idea

Could you put it as a constant in `lib/constants/app.ts`?

Something like this but it also parses the env into number if possible

```
export const APP_DOCUMENT_UPLOAD_SIZE_LIMIT = process.env.NEXT_PUBLIC_DOCUMENT_SIZE_UPLOAD_LIMIT ?? 50;
```

Comment (Mythie) [reply apps/web/src/app/(dashboard)/documents/upload-document.tsx:108]: That'll make it `string | number`, we would also need to use whatever value in the `next.config.js` for the body size limits this may also apply to certain API routes.

File: packages/ui/primitives/document-dropzone.tsx [hunk 2]
```diff
@@ -175,10 +187,10 @@ export const DocumentDropzone = ({
           </p>
 
           <p className="text-muted-foreground/80 mt-1 text-sm">
-            {disabled ? disabledMessage : 'Drag & drop your document here.'}
+            {disabled ? disabledMessage : 'Drag & drop your PDF here.'}
```
Comment (dephraiim) [inline packages/ui/primitives/document-dropzone.tsx:190]: Do we need to specify a pdf? Document is fine imo

Comment (dguyen) [reply packages/ui/primitives/document-dropzone.tsx:190]: It's done to clarify that we only take in PDFs

File: apps/web/src/app/(dashboard)/documents/upload-document.tsx (hunk 1 above)
Comment (dguyen) [reply apps/web/src/app/(dashboard)/documents/upload-document.tsx:108]: Yeah that's why I said it would require parsing

> we would also need to use whatever value in the next.config.js 

Not necessarily since this is purely for document upload limits

Comment (sumitbishti) [reply apps/web/src/app/(dashboard)/documents/upload-document.tsx:108]: do i need to use the document_size_limit env variable inside config in [trpc].ts files?
```
export const config = {
  maxDuration: 60,
  api: {
    bodyParser: {
      sizeLimit: '50mb',
    },
  },
};
```

Comment (dguyen) [reply apps/web/src/app/(dashboard)/documents/upload-document.tsx:108]: Currently the file upload transport doesn't go through TRPC, so I don't think it's relevant here

Comment (Mythie) [reply apps/web/src/app/(dashboard)/documents/upload-document.tsx:108]: It goes through server actions which have a similar limit configured. We also have a maxSize set on the dropzone primitive.

I think the best course of action would be to hardcode this but extract that current mix of magic numbers into a constant somewhere.

File: packages/lib/constants/app.ts [hunk 3]
```diff
@@ -6,3 +6,6 @@ export const APP_FOLDER = IS_APP_MARKETING ? 'marketing' : 'web';
 export const APP_BASE_URL = IS_APP_WEB
   ? process.env.NEXT_PUBLIC_WEBAPP_URL
   : process.env.NEXT_PUBLIC_MARKETING_URL;
+
+export const APP_DOCUMENT_UPLOAD_SIZE_LIMIT =
+  Number(process.env.NEXT_PUBLIC_DOCUMENT_SIZE_UPLOAD_LIMIT) ?? 50;
```
Comment (dguyen) [inline packages/lib/constants/app.ts:11]: I don't think this will work as expected

```
Number(undefined) ?? 50 // Becomes NaN
```

I think you meant something like this?

```
Number(undefined) || 50 // Becomes 50
```

File: packages/ui/primitives/document-dropzone.tsx [hunk 4]
```diff
@@ -10,6 +10,8 @@ import { megabytesToBytes } from '@documenso/lib/universal/unit-convertions';
 import { cn } from '../lib/utils';
 import { Card, CardContent } from './card';
 
+import {APP_DOCUMENT_UPLOAD_SIZE_LIMIT} from'@documenso/lib/constants/app';
```
Comment (dguyen) [inline packages/ui/primitives/document-dropzone.tsx:13]: This should be causing linting issues, is your workspace configured correctly?

I also noticed some files in this PR are having final whitespace is being removed, which shouldn't happen if the environment is correctly configured

**Review by dguyen — CHANGES_REQUESTED** [review]

File: packages/ui/primitives/document-dropzone.tsx (hunk 4 above)
Comment (sumitbishti) [reply packages/ui/primitives/document-dropzone.tsx:13]: I need help in this issue!

Comment (dguyen) [reply packages/ui/primitives/document-dropzone.tsx:13]: Try restart vscode first if you haven't yet done so

When you go to this file, it should show up as linting errors

image

Have a look in the vscode `problems` tab to see if it's working correctly

Also have a look bottom right of vscode to see if prettier is working, it should show up and have a check next to it

image

File: apps/web/src/app/(dashboard)/documents/upload-document.tsx [hunk 5]
```diff
@@ -96,13 +98,23 @@ export const UploadDocument = ({ className }: UploadDocumentProps) => {
     }
   };
 
+  const onFileDropRejected = () => {
+    toast({
+      title: `File cannot be larger than ${APP_DOCUMENT_UPLOAD_SIZE_LIMIT}mb`,
```
Comment (catalinpit) [inline apps/web/src/app/(dashboard)/documents/upload-document.tsx:103]: I'd swap the "title" message with the "description" message.

File: packages/lib/constants/app.ts [hunk 6]
```diff
@@ -6,3 +6,6 @@ export const APP_FOLDER = IS_APP_MARKETING ? 'marketing' : 'web';
 export const APP_BASE_URL = IS_APP_WEB
   ? process.env.NEXT_PUBLIC_WEBAPP_URL
   : process.env.NEXT_PUBLIC_MARKETING_URL;
+
+export const APP_DOCUMENT_UPLOAD_SIZE_LIMIT =
+  Number(process.env.NEXT_PUBLIC_DOCUMENT_SIZE_UPLOAD_LIMIT) || 50;
```
Comment (catalinpit) [inline packages/lib/constants/app.ts:11]: I don't think the empty lines from the end should be deleted

File: apps/web/src/app/(dashboard)/documents/upload-document.tsx (hunk 5 above)
Comment (catalinpit) [reply apps/web/src/app/(dashboard)/documents/upload-document.tsx:103]: Also, it should probably be `MB`

**Review by catalinpit — CHANGES_REQUESTED** [review]
Looks good, thanks for the work!

**Review by catalinpit — APPROVED** [review]

**Review by dguyen — APPROVED** [review]
